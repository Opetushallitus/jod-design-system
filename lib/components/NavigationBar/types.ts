import { ServiceVariant } from '../../utils';

export type LangCode = 'fi' | 'sv' | 'en';

export const langLabels: Record<LangCode, string> = {
  en: 'In English',
  fi: 'Suomeksi',
  sv: 'PÃ¥ svenska',
};

export type LanguageMenuLinkComponent = React.ComponentType<{
  to: string;
  onClick: () => void;
  type?: string;
  children: React.ReactNode;
  className?: string;
}>;

export interface LanguageButtonProps {
  /** Service variant */
  serviceVariant: ServiceVariant;
  /** Current language in use */
  language: LangCode;
  /** Languages that are in use from supported ones  */
  supportedLanguageCodes: LangCode[];
  generateLocalizedPath: (lng: string) => string;
  LinkComponent: LanguageMenuLinkComponent;
  translations: LanguageTranslations;
  dataTestId?: string;
}

export interface LanguageMenuProps extends LanguageButtonProps {
  onClick: () => void;
}

export type LanguageTranslations = Record<LangCode, LanguageMeta>;

export interface LanguageMeta {
  change: string;
  label: string;
}
