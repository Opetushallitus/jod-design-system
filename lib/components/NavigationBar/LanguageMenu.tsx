import { JodCircle } from '../../icons';
import { PopupList } from '../PopupList/PopupList';
import { langLabels, LanguageMenuProps } from './types';

const ListItems = ({
  onClick,
  language,
  supportedLanguageCodes,
  LinkComponent,
  generateLocalizedPath,
  dataTestId,
}: LanguageMenuProps) => {
  return supportedLanguageCodes.map((lng) => (
    <div className="ds:flex ds:flex-row ds:justify-center ds:items-center ds:gap-2 ds:w-full" key={lng}>
      <div className="ds:w-5 ds:px-2">
        {lng === language && <JodCircle size={12} className="ds:text-secondary-1" />}
      </div>
      <LinkComponent
        to={generateLocalizedPath(lng)}
        onClick={onClick}
        type="button"
        className="ds:w-full ds:text-menu ds:hover:underline ds:p-3"
        data-testid={dataTestId ? `${dataTestId}-menu-item-${lng}` : undefined}
      >
        {langLabels[lng] ?? lng}
      </LinkComponent>
    </div>
  ));
};

export const LanguageMenu = ({
  onClick,
  language,
  supportedLanguageCodes,
  generateLocalizedPath,
  LinkComponent,
  translations,
  dataTestId,
}: LanguageMenuProps) => {
  return (
    <div data-testid={dataTestId ? `${dataTestId}-menu-popup` : undefined}>
      <PopupList classNames="ds:gap-2">
        <ListItems
          onClick={onClick}
          supportedLanguageCodes={supportedLanguageCodes}
          language={language}
          generateLocalizedPath={generateLocalizedPath}
          LinkComponent={LinkComponent}
          translations={translations}
          dataTestId={dataTestId}
        />
      </PopupList>
    </div>
  );
};
