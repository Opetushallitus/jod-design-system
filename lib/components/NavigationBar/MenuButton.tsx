import { JodMenu } from '../../icons';

export interface MenuButtonProps {
  onClick: () => void;
  label: string;
}

export const MenuButton = ({ onClick, label }: MenuButtonProps) => {
  return (
    <button
      onClick={onClick}
      className="ds:flex ds:flex-col ds:md:flex-row ds:gap-2 ds:md:gap-3 ds:justify-center ds:items-center ds:select-none ds:cursor-pointer"
      data-testid="open-nav-menu"
      aria-haspopup="dialog"
    >
      <JodMenu className="ds:mx-auto" />
      <span className="ds:md:text-[14px] ds:sm:text-[12px] ds:text-[10px] ds:font-semibold">{label}</span>
    </button>
  );
};
