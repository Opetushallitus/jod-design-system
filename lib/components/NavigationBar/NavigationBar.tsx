import { useMediaQueries } from '../../hooks/useMediaQueries';

export interface NavigationBarLinkProps {
  className?: string;
  role?: string;
  title?: string;
  children: React.ReactNode;
}

export type NavigationBarLink = React.ComponentType<NavigationBarLinkProps>;

export interface NavigationBarProps {
  /** Navigation logo */
  logo: React.ReactNode;
  /** Place for menu opener button */
  menuComponent?: React.ReactNode;
  /** For language selection button **/
  onLanguageClick?: () => void;
  /** Navigation avatar */
  user?: {
    name: string;
    component: NavigationBarLink;
  };
  login: {
    url: string;
    text: string;
  };
}

/**
 * This component is a navigation bar that displays a logo, and an avatar.
 */
export const NavigationBar = ({ logo, menuComponent, onLanguageClick, user, login }: NavigationBarProps) => {
  const { sm } = useMediaQueries();

  const initials = user?.name
    .split(' ')
    .map((part) => part[0])
    .splice(0, 2)
    .join('')
    .toUpperCase();

  return (
    <div className="ds-min-w-min ds-shadow-border ds-bg-white">
      <nav
        role="navigation"
        className="ds-mx-auto ds-flex ds-h-[56px] ds-items-center ds-justify-between ds-gap-4 ds-px-7 ds-py-3 ds-font-semibold lg:ds-container"
      >
        {logo}
        <ul className="ds-inline-flex ds-items-center ds-gap-3 sm:ds-gap-5">
          {menuComponent && sm && <li>{menuComponent}</li>}
          <li className="ds-ml-5">
            <button
              className="ds-flex ds-h-8 ds-w-8 ds-items-center ds-justify-center ds-rounded-full ds-bg-bg-gray-2"
              onClick={onLanguageClick}
            >
              <span className="material-symbols-outlined size-24 ds-select-none ds-text-black">language</span>
            </button>
          </li>
          <li>
            {user ? (
              <user.component
                className="ds-flex ds-h-8 ds-w-8 ds-items-center ds-justify-center ds-rounded-full ds-bg-secondary-3 ds-text-white"
                role="img"
                title={user.name}
              >
                {initials}
              </user.component>
            ) : (
              <a
                href={login.url}
                className="ds-flex ds-h-8 ds-w-8 ds-items-center ds-justify-center ds-rounded-full ds-bg-bg-gray-2"
              >
                <span className="material-symbols-outlined size-24 ds-select-none ds-text-black">person</span>
              </a>
            )}
          </li>
          {menuComponent && !sm && <li className="ds-ml-3">{menuComponent}</li>}
        </ul>
      </nav>
    </div>
  );
};
