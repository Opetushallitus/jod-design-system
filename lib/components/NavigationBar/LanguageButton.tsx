import React, { JSX } from 'react';
import { useMediaQueries } from '../../hooks/useMediaQueries';
import { usePopupMenu } from '../../hooks/usePopupMenu';
import { JodCaretDown, JodCaretUp, JodLanguage } from '../../icons';
import { LanguageMenu } from './LanguageMenu';
import { LangCode, LanguageButtonProps, LanguageTranslations } from './types';

const getLanguageOrder = (current: LangCode, translations: LanguageTranslations): JSX.Element => {
  const orderMap: Record<LangCode, LangCode[]> = {
    fi: ['en', 'sv', 'fi'],
    sv: ['en', 'fi', 'sv'],
    en: ['fi', 'sv', 'en'],
  };
  const ordered = orderMap[current];
  return (
    <div className="ds:sr-only">
      {ordered.map((lang, idx) => (
        <React.Fragment key={lang}>
          <span lang={lang}>{translations[lang].change}</span>
          {idx < ordered.length - 1 && <br />}
        </React.Fragment>
      ))}
    </div>
  );
};

export const LanguageButton = ({
  serviceVariant,
  language,
  supportedLanguageCodes,
  generateLocalizedPath,
  LinkComponent,
  translations,
  dataTestId,
}: LanguageButtonProps) => {
  const { md, sm } = useMediaQueries();
  const { isOpen: langMenuOpen, close: closeLanguageMenu, triggerProps, menuProps } = usePopupMenu();
  const carets = md ? <>{langMenuOpen ? <JodCaretUp size={20} /> : <JodCaretDown size={20} />}</> : null;

  const positionClass = sm ? 'ds:absolute ds:right-0' : 'ds:fixed ds:left-4 ds:right-4';

  return (
    <div className="ds:relative" data-testid={dataTestId}>
      <button
        {...triggerProps}
        className="ds:flex ds:flex-col ds:md:flex-row ds:justify-center ds:items-center ds:select-none ds:cursor-pointer ds:gap-y-2 ds:gap-x-3"
        data-testid={dataTestId ? `${dataTestId}-trigger` : undefined}
      >
        <JodLanguage className="mx-auto" />
        <span className="ds:whitespace-nowrap ds:md:text-[14px] ds:sm:text-[12px] ds:text-[10px] ds:font-semibold">
          {translations[language].label}
          {getLanguageOrder(language, translations)}
        </span>
        {carets}
      </button>
      {langMenuOpen && (
        <div
          {...menuProps}
          className={`ds:z-60 ds:flex ds:justify-center ds:translate-y-8 ${positionClass}`}
          data-testid={dataTestId ? `${dataTestId}-menu` : undefined}
        >
          <LanguageMenu
            serviceVariant={serviceVariant}
            supportedLanguageCodes={supportedLanguageCodes}
            language={language}
            generateLocalizedPath={generateLocalizedPath}
            LinkComponent={LinkComponent}
            onClick={closeLanguageMenu}
            translations={translations}
            dataTestId={dataTestId}
          />
        </div>
      )}
    </div>
  );
};
