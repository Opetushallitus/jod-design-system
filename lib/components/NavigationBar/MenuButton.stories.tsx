import type { StoryObj } from '@storybook/react-vite';
import React from 'react';
import type { TitledMeta } from '../../utils';

import { fn } from 'storybook/test';
import { ServiceVariantProvider } from '../../hooks/useServiceVariant';
import { LinkComponent, NavigationMenu, NavigationMenuProps } from '../NavigationMenu';
import { externalLinkSections, languageSelectionItems, menuSection } from '../NavigationMenu/commonStoriesData';
import { MenuButton, MenuButtonProps } from './MenuButton';

const meta = {
  title: 'Navigation/MenuButton',
  component: MenuButton,
  tags: ['autodocs'],
} satisfies TitledMeta<typeof MenuButton>;

export default meta;

type Story = StoryObj<typeof meta>;

const menuProps: NavigationMenuProps = {
  portalLinkLabel: 'Osaamispolkuportaali',
  PortalLinkComponent: ({ children, className }: LinkComponent) => (
    <a href="/#" className={className}>
      {children}
    </a>
  ),
  onClose: fn(),
  open: true,
  menuSection: menuSection,
  openSubMenuLabel: 'Avaa alivalikko',
  ariaCloseMenu: 'Sulje valikko',
  externalLinkSections: externalLinkSections,
  languageSelectionItems: languageSelectionItems,
  languageSelectionTitle: 'Käyttökieli',
  selectedLanguage: 'fi',
  serviceVariant: 'yksilo',
  externalLinkIconAriaLabel: 'Linkki johtaa palvelun ulkopuolelle',
};

const DefaultRender = (props: MenuButtonProps) => {
  const [isOpen, setIsOpen] = React.useState(false);
  return (
    <ServiceVariantProvider value="yksilo">
      <MenuButton {...props} onClick={() => setIsOpen(true)} />
      <NavigationMenu {...menuProps} open={isOpen} onClose={() => setIsOpen(false)} />
    </ServiceVariantProvider>
  );
};

export const Default: Story = {
  parameters: {
    design: {
      type: 'figma',
      url: 'https://www.figma.com/design/6M2LrpSCcB0thlFDaQAI2J/cx_jod_client?node-id=14130-48895',
    },
    docs: {
      description: {
        story: 'Menu button to open the menu',
      },
    },
  },
  render: DefaultRender,
  args: {
    label: 'Valikko',
    onClick: fn(),
  },
  decorators: [
    (Story) => (
      <div className="ds:flex ds:item-center ds:justify-center ds:h-[250px]">
        <Story />
      </div>
    ),
  ],
};
