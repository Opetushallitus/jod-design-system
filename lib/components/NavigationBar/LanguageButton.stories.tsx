import type { StoryObj } from '@storybook/react-vite';
import { useRef, useState } from 'storybook/preview-api';
import type { TitledMeta } from '../../utils';

import React from 'react';
import { LanguageButton } from './LanguageButton';

const meta = {
  title: 'Navigation/LanguageButton',
  component: LanguageButton,
  tags: ['autodocs'],
} satisfies TitledMeta<typeof LanguageButton>;

export default meta;

type Story = StoryObj<typeof meta>;

const Render = (args: Story['args']) => {
  const [langMenuOpen, setLangMenuOpen] = useState(false);
  const menuRef = useRef<HTMLDivElement | null>(null);
  const handleBlur = () => setLangMenuOpen(false);
  return (
    <LanguageButton
      {...args}
      onClick={() => setLangMenuOpen(!langMenuOpen)}
      langMenuOpen={langMenuOpen}
      menuRef={menuRef}
      onMenuBlur={handleBlur}
      onMenuClick={() => setLangMenuOpen(false)}
    />
  );
};

export const Default: Story = {
  parameters: {
    design: {
      type: 'figma',
      url: 'https://www.figma.com/design/6M2LrpSCcB0thlFDaQAI2J/cx_jod_client?node-id=17454-164370',
    },
    docs: {
      description: {
        story: 'Language button component for switching languages.',
      },
    },
  },
  args: {
    supportedLanguageCodes: ['fi', 'sv', 'en'],
    translations: {
      fi: { change: 'Vaihda kieli.', label: 'Suomeksi' },
      sv: { change: 'Andra språk.', label: 'På svenska' },
      en: { change: 'Change language.', label: 'In English' },
    },
    onClick: () => {},
    langMenuOpen: false,
    menuRef: React.createRef(),
    onMenuBlur: () => {},
    onMenuClick: () => {},
    language: 'fi',
    generateLocalizedPath: (code: string) => `/${code}`,
    LinkComponent: ({ children, className, ...rest }) => (
      <a href="/#" className={className} {...rest}>
        {children}
      </a>
    ),
  },
  decorators: [
    (Story) => (
      <div className="ds:flex ds:item-center ds:justify-center ds:h-[250px]">
        <Story />
      </div>
    ),
  ],
  render: Render,
};
