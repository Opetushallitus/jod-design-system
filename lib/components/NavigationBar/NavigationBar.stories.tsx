import type { Meta, StoryObj } from '@storybook/react';

import { NavigationBar, type NavigationBarProps } from './NavigationBar';

const meta = {
  component: NavigationBar,
  tags: ['autodocs'],
} satisfies Meta<typeof NavigationBar>;

export default meta;

type Story = StoryObj<typeof meta>;

const parameters = {
  design: {
    type: 'figma',
    url: 'https://www.figma.com/file/6M2LrpSCcB0thlFDaQAI2J/cx_jod_client?node-id=542%3A8159',
  },
  layout: 'fullscreen',
  viewport: {
    defaultViewport: 'desktop',
  },
};

const items: NavigationBarProps['items'] = [
  {
    text: 'YksilÃ¶ille',
    active: true,
    component: ({ children, ...rootProps }) => (
      <a href="/yksiloille" {...rootProps}>
        {children}
      </a>
    ),
  },
  {
    text: 'Ohjaajille',
    active: false,
    component: ({ children, ...rootProps }) => (
      <a href="/ohjaajille" {...rootProps}>
        {children}
      </a>
    ),
  },
  {
    text: 'Kouluttajille',
    active: false,
    component: ({ children, ...rootProps }) => (
      <a href="/kouluttajille" {...rootProps}>
        {children}
      </a>
    ),
  },
  {
    text: 'Viranomaisille',
    active: false,
    component: ({ children, ...rootProps }) => (
      <a href="/viranomaisille" {...rootProps}>
        {children}
      </a>
    ),
  },
];

const user: NavigationBarProps['user'] = {
  name: 'Jane Doe',
  component: ({ children, ...rootProps }) => (
    <a href="/profiili" aria-label="Profiili" {...rootProps}>
      {children}
    </a>
  ),
};

const logo = (
  <a href="/" className="flex">
    <div className="inline-flex select-none items-center gap-4 text-[24px] leading-[140%] text-accent">
      <div className="h-8 w-8 bg-accent"></div>JOD
    </div>
  </a>
);

const args: NavigationBarProps = {
  logo,
  items,
  user,
};

export const Default: Story = {
  parameters: {
    ...parameters,
    docs: {
      description: {
        story: 'This story demonstrates how the navigation bar looks with navigation items and an avatar.',
      },
    },
  },
  args,
};

export const Plain: Story = {
  parameters: {
    ...parameters,
    docs: {
      description: {
        story: 'This story demonstrates how the navigation bar looks without any items or an avatar.',
      },
    },
  },
  args: {
    logo,
  },
};

export const Items: Story = {
  parameters: {
    ...parameters,
    docs: {
      description: {
        story: 'This story demonstrates how the navigation bar looks with navigation items.',
      },
    },
  },
  args: {
    logo,
    items,
  },
};

export const Avatar: Story = {
  parameters: {
    ...parameters,
    docs: {
      description: {
        story: 'This story demonstrates how the navigation bar looks with an avatar.',
      },
    },
  },
  args: {
    logo,
    user,
  },
};

export const Sticky: Story = {
  decorators: [
    (Story) => (
      <div>
        <header className="sticky top-0">
          <Story />
        </header>
        <main>
          {[...Array<undefined>(20)].map((_p, index) => (
            <p key={index} className="p-4">
              Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et
              dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex
              ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu
              fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt
              mollit anim id est laborum.
            </p>
          ))}
        </main>
      </div>
    ),
  ],
  parameters: {
    ...parameters,
    docs: {
      description: {
        story: 'This story demonstrates how the navigation bar behaves when it is sticky.',
      },
      story: {
        inline: false,
        iframeHeight: 400,
      },
      source: {
        code: '<header className="sticky top-0">\n  <NavigationBar />\n</header>',
      },
    },
  },
  args,
};
