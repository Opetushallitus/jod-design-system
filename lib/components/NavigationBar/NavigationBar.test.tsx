import '@testing-library/jest-dom';
import { render, screen, waitFor } from '@testing-library/react';
import { describe, expect, it } from 'vitest';
import { NavigationBar, NavigationBarLinkProps } from './NavigationBar';

describe('NavigationBar', () => {
  const user = {
    name: 'Reetta Räppänä',
    component: ({ children, ...rootProps }: NavigationBarLinkProps) => (
      <a href="/profile" aria-label="Profile" {...rootProps}>
        {children}
      </a>
    ),
  };

  it('renders no navigation items and no user', () => {
    const { container } = render(
      <NavigationBar
        renderLink={({ children }) => <div>{children}</div>}
        logo={{ to: '/', language: 'fi', srText: 'jod' }}
      />,
    );

    // Assert snapshot
    expect(container.firstChild).toMatchSnapshot();

    // Assert user is not rendered
    const userAvatar = screen.queryByTitle(user.name);
    expect(userAvatar).toBeNull();
  });

  it('renders with serviceBar and handles scrolling', async () => {
    const contentText = 'Service Bar Content';
    render(
      <NavigationBar
        renderLink={({ children }) => <div>{children}</div>}
        logo={{ to: '/', language: 'fi', srText: 'jod' }}
        showServiceBar
        serviceBarVariant="yksilo"
        serviceBarTitle="Osaamispolkuni"
        serviceBarContent={<div>{contentText}</div>}
      />,
    );

    const title = screen.getByText('Osaamispolkuni');
    expect(title).toBeInTheDocument();
    expect(title).toBeVisible();

    const content = screen.getByText(contentText);
    expect(content).toBeInTheDocument();
    expect(content).toBeVisible();

    // Simulate scroll
    window.scrollY = 100;
    window.dispatchEvent(new Event('scroll'));

    // Wait for DOM update and assert elements are removed
    await waitFor(() => {
      expect(screen.queryByText('Osaamispolkuni')).not.toBeInTheDocument();
      expect(screen.queryByText(contentText)).not.toBeInTheDocument();
    });
  });
});

it('does not toggle serviceBar contents multiple times during animation pending', async () => {
  const contentText = 'Service Bar Content';
  render(
    <NavigationBar
      renderLink={({ children }) => <div>{children}</div>}
      logo={{ to: '/', language: 'fi', srText: 'jod' }}
      showServiceBar
      serviceBarVariant="yksilo"
      serviceBarTitle="Osaamispolkuni"
      serviceBarContent={<div>{contentText}</div>}
    />,
  );

  // Scroll down
  window.scrollY = 100;
  window.dispatchEvent(new Event('scroll'));

  // Immediately try to scroll again before timeout
  window.scrollY = 200;
  window.dispatchEvent(new Event('scroll'));

  // Wait for DOM update
  await waitFor(() => {
    expect(screen.queryByText('Osaamispolkuni')).not.toBeInTheDocument();
    expect(screen.queryByText(contentText)).not.toBeInTheDocument();
  });

  // Scroll back to top after animation pending
  await new Promise((r) => setTimeout(r, 250));
  window.scrollY = 0;
  window.dispatchEvent(new Event('scroll'));

  await waitFor(() => {
    expect(screen.getByText('Osaamispolkuni')).toBeVisible();
    expect(screen.getByText(contentText)).toBeVisible();
  });
});
