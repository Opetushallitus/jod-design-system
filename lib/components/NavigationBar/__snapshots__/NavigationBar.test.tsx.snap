// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`NavigationBar > renders no navigation items and no user 1`] = `
<div
  class="ds-min-w-min ds-shadow-border ds-bg-white"
>
  <nav
    class="ds-mx-auto ds-flex ds-h-[56px] ds-items-center ds-justify-between ds-gap-4 ds-px-7 ds-py-3 ds-font-semibold lg:ds-container"
    role="navigation"
  >
    <div>
      logo
    </div>
    <ul
      class="ds-inline-flex ds-items-center ds-gap-3 sm:ds-gap-5"
    >
      <li
        class="ds-ml-5"
      >
        <button
          class="ds-flex ds-h-8 ds-w-8 ds-items-center ds-justify-center ds-rounded-full ds-bg-bg-gray-2"
        >
          <span
            class="material-symbols-outlined size-24 ds-select-none ds-text-black"
          >
            language
          </span>
        </button>
      </li>
      <li>
        <a
          class="ds-flex ds-h-8 ds-w-8 ds-items-center ds-justify-center ds-rounded-full ds-bg-bg-gray-2"
          href="/login"
        >
          <span
            class="material-symbols-outlined size-24 ds-select-none ds-text-black"
          >
            person
          </span>
        </a>
      </li>
    </ul>
  </nav>
</div>
`;

exports[`NavigationBar > renders only user 1`] = `
<div
  class="ds-min-w-min ds-shadow-border ds-bg-white"
>
  <nav
    class="ds-mx-auto ds-flex ds-h-[56px] ds-items-center ds-justify-between ds-gap-4 ds-px-7 ds-py-3 ds-font-semibold lg:ds-container"
    role="navigation"
  >
    <div>
      logo
    </div>
    <ul
      class="ds-inline-flex ds-items-center ds-gap-3 sm:ds-gap-5"
    >
      <li
        class="ds-ml-5"
      >
        <button
          class="ds-flex ds-h-8 ds-w-8 ds-items-center ds-justify-center ds-rounded-full ds-bg-bg-gray-2"
        >
          <span
            class="material-symbols-outlined size-24 ds-select-none ds-text-black"
          >
            language
          </span>
        </button>
      </li>
      <li>
        <a
          aria-label="Profile"
          class="ds-flex ds-h-8 ds-w-8 ds-items-center ds-justify-center ds-rounded-full ds-bg-secondary-3 ds-text-white"
          href="/profile"
          role="img"
          title="Reetta Räppänä"
        >
          RR
        </a>
      </li>
    </ul>
  </nav>
</div>
`;
